hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -126,7 +134,52 @@ public void handle(@NonNull final HandleContext context) throws HandleException
     @NonNull
     @Override
     public Fees calculateFees(final FeeContext feeContext) {
-        // Todo: Implement the logic for calculating fees
-        return Fees.FREE;
+        final var body = feeContext.body();
+        final var op = body.tokenRejectOrThrow();
+        final var config = feeContext.configuration();
+        final var tokenMultiplier =
+                requireNonNull(config).getConfigData(FeesConfig.class).tokenTransferUsageMultiplier();
+
+        final int totalRejections = op.rejections().size();
+        int numOfFungibleTokenRejections = 0;
+        int numOfNFTRejections = 0;
+        for (final var rejection : op.rejections()) {
+            if (rejection.hasFungibleToken()) {
+                numOfFungibleTokenRejections++;
+            } else {
+                numOfNFTRejections++;
+            }
+        }
+
+        final int weightedTokensInvolved = tokenMultiplier * totalRejections;
+        final int weightedFungibleTokens = tokenMultiplier * numOfFungibleTokenRejections;
+
+        final long bpt =
+                calculateBytesPerTransaction(weightedTokensInvolved, weightedFungibleTokens, numOfNFTRejections);
```

Comment:
It is extremely important and useful to add unit tests and E2E tests in the same PR, to validate the correctness of the code.

Reply:
I agree, I will close this PR and open one which contains the whole logic.